### ISSUE TITLE
styled first parameter needs extended typing

### ISSUE THREAD

**Issue by dugip** [macaron-css/macaron#35]
As it stands right now, you can't style components that use a `forwardRef` wrapper, which forced me to cast them to any.

Example

```
const MyForwardRefComponent = React.forwardRef((props, ref) => <h1 ref={ref}>Hello World</h1>);

// does not work
const StyledMyForwardRefComponent1 = styled(MyForwardRefComponent, {...});

// does work
const StyledMyForwardRefComponent2 = styled(MyForwardRefComponent as any, {...});
```

Would appreciate if this received a fix.

**Comment by Mokshit06** [comment]
Will fix this!

**Comment by Mokshit06** [comment]
Should be fixed in v1.2.1. Thanks for reporting.
